Setup and Orientation
# Data_Check.ipynb
## IIT414W · Unit I · 11/03/2026

In [47]:
# ── Dependency Guard ───────────────────────────────────────────────
import importlib, subprocess, sys

_REQUIRED = {
    'numpy': 'numpy',
    'pandas': 'pandas',
    'sklearn': 'scikit-learn',
    'matplotlib': 'matplotlib',
    'seaborn': 'seaborn',
    'fastf1': 'fastf1',
}

_missing = []
for _mod, _pip in _REQUIRED.items():
    try:
        importlib.import_module(_mod)
    except ModuleNotFoundError:
        _missing.append(_pip)

if _missing:
    print(f'Installing missing packages: {_missing}')
    subprocess.check_call([sys.executable, '-m', 'pip', 'install', '--quiet'] + _missing)
    print('Done. Packages installed successfully.')
else:
    print('All required packages already installed ✓')

# ── Library Imports ───────────────────────────────────────────────
import os                        # Working directory checks
import subprocess                # Git command checks
import importlib                 # Runtime dependency checks
import numpy as np               # Numeric support
import pandas as pd              # Tables and diagnostics
import matplotlib.pyplot as plt  # Plotting
import seaborn as sns            # Statistical plotting
import fastf1                    # Formula 1 data access

print(f'fastf1  : {fastf1.__version__}')
print(f'pandas  : {pd.__version__}')

All required packages already installed ✓
fastf1  : 3.8.1
pandas  : 2.3.3


In [48]:
# ── Reproducibility Header ────────────────────────────────────────────

import sys, random
import numpy as np
import warnings

RANDOM_SEED = 414  # Course constant. Do not change.
random.seed(RANDOM_SEED)
np.random.seed(RANDOM_SEED)
warnings.filterwarnings('ignore', category=FutureWarning)

# Environment check
print(f'Python  : {sys.version.split()[0]}')
print(f'NumPy   : {np.__version__}')
print(f'Seed    : {RANDOM_SEED}')

Python  : 3.12.3
NumPy   : 2.4.2
Seed    : 414


### FastF1 Session Check

In [49]:
# Session load
session = fastf1.get_session(2024, 'Abu Dhabi', 'Q')
session.load(laps=True, telemetry=False, weather=False, messages=False)

core           INFO 	Loading data for Abu Dhabi Grand Prix - Qualifying [v3.8.1]
req            INFO 	Using cached data for session_info
req            INFO 	Using cached data for driver_info
req            INFO 	Using cached data for session_status_data
req            INFO 	Using cached data for track_status_data
req            INFO 	Using cached data for _extended_timing_data
req            INFO 	Using cached data for timing_app_data
core           INFO 	Processing timing data...
core           INFO 	Finished loading data for 20 drivers: ['4', '81', '55', '27', '1', '10', '63', '14', '77', '11', '22', '30', '18', '16', '20', '23', '24', '44', '43', '61']


In [50]:
print('Session Event:', session.event["EventName"], '\nSession Name: ', session.name, )
print('Numbers of laps:', len(session.laps))
print('Results shape:', session.results.shape)
print('Results columns:', list(session.results.columns))
session.laps.head()

Session Event: Abu Dhabi Grand Prix 
Session Name:  Qualifying
Numbers of laps: 261
Results shape: (20, 22)
Results columns: ['DriverNumber', 'BroadcastName', 'Abbreviation', 'DriverId', 'TeamName', 'TeamColor', 'TeamId', 'FirstName', 'LastName', 'FullName', 'HeadshotUrl', 'CountryCode', 'Position', 'ClassifiedPosition', 'GridPosition', 'Q1', 'Q2', 'Q3', 'Time', 'Status', 'Points', 'Laps']


,Time,Driver,DriverNumber,LapTime,LapNumber,Stint,PitOutTime,PitInTime,Sector1Time,Sector2Time,...,FreshTyre,Team,LapStartTime,LapStartDate,TrackStatus,Position,Deleted,DeletedReason,FastF1Generated,IsAccurate
0,0 days 00:22:41.925000,NOR,4,NaT,1.0,1.0,0 days 00:20:03.650000,NaT,NaT,0 days 00:00:49.350000,...,True,McLaren,0 days 00:20:03.650000,NaT,1,NaN,None,,False,False
1,0 days 00:24:05.607000,NOR,4,0 days 00:01:23.682000,2.0,1.0,NaT,NaT,0 days 00:00:17.129000,0 days 00:00:36.301000,...,True,McLaren,0 days 00:22:41.925000,NaT,1,NaN,None,,False,True
2,0 days 00:25:57.551000,NOR,4,0 days 00:01:51.944000,3.0,1.0,NaT,0 days 00:25:56.445000,0 days 00:00:21.676000,0 days 00:00:47.997000,...,True,McLaren,0 days 00:24:05.607000,NaT,1,NaN,None,,False,False
3,0 days 00:31:52.979000,NOR,4,NaT,4.0,2.0,0 days 00:29:10.623000,NaT,NaT,0 days 00:00:46.011000,...,True,McLaren,0 days 00:25:57.551000,NaT,1,NaN,None,,False,False
4,0 days 00:33:25.776000,NOR,4,0 days 00:01:32.797000,5.0,2.0,NaT,0 days 00:33:24.624000,0 days 00:00:17.017000,0 days 00:00:36.024000,...,True,McLaren,0 days 00:31:52.979000,NaT,1,NaN,None,,False,False


## Jolpica API Query

In [51]:
import requests

url = 'https://api.jolpi.ca/ergast/f1/2024/driverstandings/'
response = requests.get(url)

In [53]:
print('HTTP Status Code:', response.status_code)
print('Content Length:', len(response.content))
drivers = response.json()["MRData"]["StandingsTable"]["StandingsLists"][0]["DriverStandings"]
print('Driver - Nationality - Points')
for d in drivers[:3]:
    print(d["Driver"]["familyName"], '-', d["Driver"]["nationality"], '-', d["points"])

HTTP Status Code: 200
Content Length: 10474
Driver - Nationality - Points
Verstappen - Dutch - 437
Norris - British - 374
Leclerc - Monegasque - 356


## Data Shape Summary

A) I Used the fast session Abu Dhabi from 2024 because its a recent race  
B) Fast1 Shape (20, 22) and Jolpica isnt a dataset so it doest have "shape" but it does has "len" which is 10474
C) there was no surprise and nothing particularly informative. The work this week was more focus on the reproducion of work enviroments than the data itself